File: hazelnet-community/src/main/kotlin/io/hazelnet/community/data/premium/DiscordServerPremiumInfo.kt
package io.hazelnet.community.data.premium

import java.util.*

data class DiscordServerPremiumInfo(
    val totalDelegation: Long,
    val maxDelegation: Long,
    val monthlyCost: Long,
    val actualMonthlyCost: Long,
    val guildMemberCount: Int,
    val remainingBalance: Long,
    val lastBillingGuildMemberCount: Int,
    val lastBillingTime: Date?,
    val lastBillingAmount: Long,
    val premiumUntil: Date?,
    val currentPremium: Boolean,
)

File: vibrantnet/core/src/types/premium/discordServerPremiumInfo.ts
export type DiscordServerPremiumInfo = {
  totalDelegation: number
  maxDelegation: number
  monthlyCost: number
  actualMonthlyCost: number
  guildMemberCount: number
  remainingBalance: number
  lastBillingGuildMemberCount: number
  lastBillingTime?: number
  lastBillingAmount: number
  premiumUntil?: number
  currentPremium: boolean
};
